modules/fedimint-dummy-client/src/db.rs

```diff
@@ -57,5 +63,41 @@ pub async fn migrate_to_v1(dbtx: &mut DatabaseTransaction<'_>) -> anyhow::Result
         warn!("Dummy client did not have client funds, skipping database migration");
     }
 
-    Ok(())
+    Ok(None)
+}
+
+pub async fn migrate_to_v2(
+    module_instance_id: ModuleInstanceId,
+    active_states: Vec<DynState>,
+    inactive_states: Vec<DynState>,
```

Comment:
I don't like that I needed to pass `DynState` here instead of something like `T::States`. But I couldn't figure out how to add a generic to the `Fn` in the `ClientMigrationMap`. @dpc or @Maan2003 if you guys have any rust pointers here :)

Reply:
Maybe this is sufficient for now

Reply:
you will have to use trait instead of `Fn` in that case, but this is fine being untyped

Reply:
I see. Right now that seems like a good improvement but maybe in a separate PR.

One unfortunate side affect of this being untyped though is that the higher layer does not know which state is applicable to which module. So all of the states are passed into each module, and each module is responsible for handling states that it doesn't know about.

Since the migration function expects the entire set of states to be returned back, a malicious module could actually drop states that belong to other modules. This is definitely not ideal and I think the proper fix is to make this typed.

Reply:
At very least filtering by `dyn_state.module_instance_id()` should be trivial.

Generic down and up-casting is definitely doable as well, but could be done as a next step.

Reply:
I added filtering logic by `module_instance_id` in `get_active_states` and the other functions, so this isn't as much as issue. Modules should only receive states from their state machines now. I'll leave making this typed for a different PR